restcomm/restcomm.mscontrol.mms/src/main/java/org/restcomm/connect/mscontrol/mms/MmsBridgeController.java

```diff
@@ -247,6 +254,19 @@ public void onReceive(Object message) throws Exception {
             onEndpointStateChanged((EndpointStateChanged) message, self, sender);
         } else if (MediaResourceBrokerResponse.class.equals(klass)) {
             onMediaResourceBrokerResponse((MediaResourceBrokerResponse<?>) message, self, sender);
+        } else if (MediaGroupResponse.class.equals(klass)) {
+            onMediaGroupResponse((MediaGroupResponse)message);
+        }
+    }
+
+    private void onMediaGroupResponse(MediaGroupResponse message) throws TransitionNotFoundException, TransitionFailedException, TransitionRollbackException {
+        if (logger.isInfoEnabled()) {
+            logger.info("Received MgcpGroupResponse: "+message.toString());
+        }
+        if (recording && stoppingStatePending) {
+            recording = Boolean.FALSE;
```

Comment:
shoudlnt this flags be set with appriate single condition,instead of composite condition like:
`
if (recording && stoppingStatePending) fsm.trastion(message,stopping);
if (recording) recording =false;
if(stoppingStatePending) stoppingStatePending = false;
`